### ISSUE TITLE
💅 noSvgWithoutTitle should report diagnostics some cases

### ISSUE THREAD

**Issue by mehm8128** [biomejs/biome#8422]
### Environment information

```bash
I've not checked on my local environment, but I've checked the source code of Biome directly and happended on the [playground](https://biomejs.dev/playground/?lineWidth=120&lintRules=noSvgWithoutTitle&code=PAA%2BAAoAIAAgAHsALwAqACAAMQAuACAAcwBoAG8AdQBsAGQAIAByAGUAcABvAHIAdAAgAGQAaQBhAGcAbgBvAHMAdABpAGMAIAAqAC8AfQAKADwAcwB2AGcAPgAKAAkAPAByAGUAYwB0ACAALwA%2BAAoACQA8AHIAZQBjAHQAIAAvAD4ACgAJADwAZwA%2BAAoACQAJADwAYwBpAHIAYwBsAGUAIAAvAD4ACgAJAAkAPABjAGkAcgBjAGwAZQAgAC8APgAKAAkACQA8AGcAPgAKAAkACQAJADwAdABpAHQAbABlAD4AUABhAHMAcwA8AC8AdABpAHQAbABlAD4ACgAJAAkACQA8AGMAaQByAGMAbABlACAALwA%2BAAoACQAJAAkAPABjAGkAcgBjAGwAZQAgAC8APgAKAAkACQA8AC8AZwA%2BAAoACQA8AC8AZwA%2BAAoAPAAvAHMAdgBnAD4ACgAKACAAIAB7AC8AKgAgADIALgAgAHMAaABvAHUAbABkACAAcgBlAHAAbwByAHQAIABkAGkAYQBnAG4AbwBzAHQAaQBjACAAKgAvAH0ACgA8AHMAdgBnACAAcgBvAGwAZQA9ACIAZwByAGEAcABoAGkAYwBzAC0AcwB5AG0AYgBvAGwAIgA%2BADwAcgBlAGMAdAAgAC8APgA8AC8AcwB2AGcAPgAKADwAcwB2AGcAIAByAG8AbABlAD0AIgBnAHIAYQBwAGgAaQBjAHMALQBzAHkAbQBiAG8AbAAgAGkAbQBnACIAPgA8AHIAZQBjAHQAIAAvAD4APAAvAHMAdgBnAD4ACgAKADwALwA%2BAA%3D%3D).
```

### Rule name

noSvgWithoutTitle

### Playground link

https://biomejs.dev/playground/?lineWidth=120&lintRules=noSvgWithoutTitle&code=PAA%2BAAoAPABzAHYAZwA%2BAAoACQA8AHIAZQBjAHQAIAAvAD4ACgAJADwAcgBlAGMAdAAgAC8APgAKAAkAPABnAD4ACgAJAAkAPABjAGkAcgBjAGwAZQAgAC8APgAKAAkACQA8AGMAaQByAGMAbABlACAALwA%2BAAoACQAJADwAZwA%2BAAoACQAJAAkAPAB0AGkAdABsAGUAPgBQAGEAcwBzADwALwB0AGkAdABsAGUAPgAKAAkACQAJADwAYwBpAHIAYwBsAGUAIAAvAD4ACgAJAAkACQA8AGMAaQByAGMAbABlACAALwA%2BAAoACQAJADwALwBnAD4ACgAJADwALwBnAD4ACgA8AC8AcwB2AGcAPgAKAAoAPABzAHYAZwAgAHIAbwBsAGUAPQAiAGcAcgBhAHAAaABpAGMAcwAtAHMAeQBtAGIAbwBsACIAPgA8AHIAZQBjAHQAIAAvAD4APAAvAHMAdgBnAD4ACgAKADwAcwB2AGcAIAByAG8AbABlAD0AIgBnAHIAYQBwAGgAaQBjAHMALQBzAHkAbQBiAG8AbAAgAGkAbQBnACIAPgA8AHIAZQBjAHQAIAAvAD4APAAvAHMAdgBnAD4ACgAKADwALwA%2BAA%3D%3D

### Expected result

When I worked on porting this rule from JSX to HTML, I found two problems.
https://github.com/biomejs/biome/pull/8412

Maybe I'll fix them by myself.
 
1. `<title>` element should not be in the descendant of other svg elements such as `<g>` for the spec of  [SVG 1.1](https://www.w3.org/TR/SVG11/struct.html#DescriptionAndTitleElements), so [the first Valid case](https://github.com/biomejs/biome/pull/8412/files#diff-12b71a5df08d3440f897fd00650536571fefb9e4d920731872a1a4407ed24fe9R48-R62) should report diagnostic. For the spec of SVG2, this is not applicable but [MDN](https://developer.mozilla.org/en-US/docs/Web/SVG/Reference/Element/title) mentions this from the point of backward compatibility.>Note: For backward compatibility with SVG 1.1, <title> elements should be the first child element of their parent.
 
2.  `<svg role="graphics-symbol"><rect /></svg>` and `<svg role="graphics-symbol img"><rect /></svg>` should report diagnostics because they don't have accessible name. [WAI-ARIA Graphics Module](https://www.w3.org/TR/2018/REC-graphics-aria-1.0-20181002/#graphics-symbol) says `graphics-symbol` role requires accessible name, and of course `img` role does.Their cases are handled in the code `_ => None,`, so we should handle them with match arms.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by mehm8128** [comment]
done with https://github.com/biomejs/biome/pull/8556
